exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/ColumnType.kt

```diff
@@ -205,6 +205,10 @@ internal fun IColumnType<*>.rawSqlType(): IColumnType<*> = when {
     else -> this
 }
 
+fun IColumnType<*>.isEntityIdentifier(): Boolean {
+    return this is EntityIDColumnType<*> && (idColumn.table as IdTable<*>).id == idColumn
+}
+
```

Comment:
I so far haven't found a reason to use this is the dao module, so I will switch to internal modifier.